desktop_gui.py

```diff
@@ -151,11 +175,23 @@ def switch_conversation(self):
         
         # add the chat history to the chat history tree view
         # call the populate branch tree function with the selected_conversation_id
-        self.populate_branch_tree(selected_conversation_id)
+        if chat_history_enabled == True:
+            self.populate_branch_tree(selected_conversation_id)
         # set the global conversation id variable to the selected conversation id
         conversation_id = selected_conversation_id
         
-
+    def warn(self, message):
+            msg = QtWidgets.QMessageBox()
+            msg.setIcon(QtWidgets.QMessageBox.Warning)
+            if message == "no model loaded":
+                msg.setText("The request failed. this is likely due to the model not being loaded. this module requires the "
+            + model + " api to be running. please check that the kobold api is running and try again.")
+            if message == "blank":
+                msg.setText("The request failed. this is likely due an error causing the response from the model to be blank.")
+            msg.setWindowTitle("Warning")
+            msg.setStandardButtons(QtWidgets.QMessageBox.Ok)
+            msg.exec_()
+            return
```

Comment:
Here would be a good spot to add logging, pretty much exactly as it's being written to the app